### ISSUE TITLE
Support mapping of JDK 16+ records to classes.

### ISSUE THREAD

**Issue by marcus-talbot42** [42BV/beanmapper#149]
Records use a different style of getter-methods, specifically, they use the exact name of the parameter passed to the record's definition. While it is possible to map records to classes if you write your own beanmapper-compliant getters, that defeats at least some of the purpose of records.

By using the following the record and tests, we receive different exceptions based on the strictness set in the BeanMapper.

```Java
public record PersonRecord(String name, String place) {

}
```

```Java
@Test
void mapPersonRecordToPersonResultStrict() {
    beanMapper = new BeanMapperBuilder().setApplyStrictMappingConvention(true).build();
    PersonRecord personRecord = new PersonRecord("Henk", "Zoetermeer");
    PersonResult personResult = beanMapper.map(personRecord, PersonResult.class);
    assertEquals(personRecord.name(), personResult.name);
    assertEquals(personRecord.place(), personResult.place);
}

@Test
void mapPersonRecordToPersonResultNotStrict() {
    beanMapper = new BeanMapperBuilder().setApplyStrictMappingConvention(false).build();
    PersonRecord personRecord = new PersonRecord("Henk", "Zoetermeer");
    PersonResult personResult = beanMapper.map(personRecord, PersonResult.class);
    assertEquals(personRecord.name(), personResult.name);
    assertEquals(personRecord.place(), personResult.place);
}
```

<img width="1336" alt="Screenshot 2022-09-16 at 16 24 39" src="https://user-images.githubusercontent.com/112508874/190662133-6b79433e-7089-4a84-b86f-1f07e878d99a.png">
<img width="978" alt="Screenshot 2022-09-16 at 16 25 21" src="https://user-images.githubusercontent.com/112508874/190662245-10e9fdca-53ff-4c3f-a281-e2ed67fa8055.png">

As stated earlier, these tests will pass if you write getters for them like so:

```Java
public record PersonRecord(String name, String place) {

    public String getName() {
        return this.name;
    }

    public String getPlace() {
        return this.place;
    }

}
```

Records are somewhat likely to become commonplace, so this would be a worthwhile addition to BeanMapper.

**Comment by marcus-talbot42** [comment]
It appears that mapping from class to an existing record-type is already possible, as long as the record is annotated with @BeanConstruct, containing the relevant field-names.

`@BeanConstruct({"name", "place"})
public record PersonRecord(String name, String place) {

}`

```
public class PersonResult {

    public String name;

    public String place;

}
```

```
@Test
    void testMapPersonResultToPersonRecord() {
        PersonResult personResult = new PersonResult();
        personResult.name = "Henk";
        personResult.place = "Zoetermeer";

        PersonRecord personRecord = beanMapper.map(personResult, PersonRecord.class);

        assertEquals("Henk", personRecord.name());
        assertEquals("Zoetermeer", personRecord.place());
    }
```

**Comment by marcus-talbot42** [comment]
Implemented in dev/4.1.0
